security/pkg/server/ca/server.go

```diff
@@ -48,35 +48,41 @@ type Server struct {
 	hostnames      []string
 	forCA          bool
 	port           int
+	monitoring     monitoringMetrics
 }
 
 // HandleCSR handles an incoming certificate signing request (CSR). It does
 // proper validation (e.g. authentication) and upon validated, signs the CSR
 // and returns the resulting certificate. If not approved, reason for refusal
 // to sign is returned as part of the response object.
 func (s *Server) HandleCSR(ctx context.Context, request *pb.CsrRequest) (*pb.CsrResponse, error) {
+	s.monitoring.CSR.Inc()
```

Comment:
Is promethus counter thread safe? otherwise, server needs to have mutex to avoid racing.

Reply:
That's a good point. I looked it up and found:
"All exported functions and methods are safe to be used concurrently unless specified otherwise."
https://godoc.org/github.com/prometheus/client_golang/prometheus
That's good news :)